includes/api/class-wc-rest-product-reviews-controller.php

```diff
@@ -143,6 +241,190 @@ public function get_item( $request ) {
 		return $response;
 	}
 
+
+	/**
+	 * Create a product review.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function create_item( $request ) {
+		$product = get_post( (int) $request['product_id'] );
+
+		if ( empty( $product->post_type ) || 'product' !== $product->post_type ) {
+			return new WP_Error( 'woocommerce_rest_product_invalid_id', __( 'Invalid product ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		if ( empty( $request['review'] ) ) {
+			return new WP_Error( 'woocommerce_rest_product_review_invalid_review', __( 'Product review content is required.', 'woocommerce' ), array( 'status' => 400 ) );
+		}
+
+		if ( empty( $request['name'] ) ) {
+			return new WP_Error( 'woocommerce_rest_product_review_invalid_name', __( 'Product review author name is required.', 'woocommerce' ), array( 'status' => 400 ) );
+		}
+
+		if ( empty( $request['email'] ) ) {
+			return new WP_Error( 'woocommerce_rest_product_review_invalid_email', __( 'Product review author email is required.', 'woocommerce' ), array( 'status' => 400 ) );
+		}
+
+		$data = array(
+			'comment_post_ID'      => $product->ID,
+			'comment_author'       => $request['name'],
+			'comment_author_email' => $request['email'],
+			'comment_content'      => $request['review'],
+			'comment_approved'     => 1,
+			'comment_type'         => 'review',
+		);
+		$product_review_id = wp_insert_comment( $data );
+		update_comment_meta( $product_review_id, 'rating', ( ! empty( $request['rating'] ) ? $request['rating'] : '0' ) );
+
+		$comment = get_comment( $product_review_id );
+		$this->update_additional_fields_for_object( $comment, $request );
+
+		/**
+		 * Fires after a single item is created or updated via the REST API.
+		 *
+		 * @param WP_Comment         $comment      Inserted object.
+		 * @param WP_REST_Request $request   Request object.
+		 * @param boolean         $creating  True when creating item, false when updating.
+		 */
+		do_action( "woocommerce_rest_insert_product_review", $comment, $request, true );
+
+		$request->set_param( 'context', 'edit' );
+		$response = $this->prepare_item_for_response( $comment, $request );
+		$response = rest_ensure_response( $response );
+		$response->set_status( 201 );
+		$base = str_replace( '(?P<product_id>[\d]+)', $product->id, $this->rest_base );
+		$response->header( 'Location', rest_url( sprintf( '/%s/%s/%d', $this->namespace, $base, $product_review_id ) ) );
+
+		return $response;
+	}
+
+	/**
+	 * Update a single product review.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_item( $request ) {
+		$id      = (int) $request['id'];
+		$product = get_post( (int) $request['product_id'] );
```

Comment:
If you don't need the full object here, might be cleaner to just compare `'product' === get_post_type( $id )`